AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine.swift

```diff
@@ -79,120 +89,101 @@ class RemoteSyncEngine: RemoteSyncEngineBehavior {
+            break
+
+        case .cleanup(let amplifyError):
+            //todo
+            print("error: \(amplifyError)")
 
-        self.api = api
+        }
+    }
 
-        guard let storageAdapter = storageAdapter else {
+    func start(api: APICategoryGraphQLBehavior = Amplify.API) {
+        self.api = api
+        guard storageAdapter != nil else {
             log.error(error: DataStoreError.nilStorageAdapter())
             remoteSyncTopicPublisher.send(completion: .failure(DataStoreError.nilStorageAdapter()))
             return
         }
-        remoteSyncTopicPublisher.send(.storageAdapterAvailable)
-
-        let pauseSubscriptionsOp = CancelAwareBlockOperation {
-            self.pauseSubscriptions()
-        }
-
-        let pauseMutationsOp = CancelAwareBlockOperation {
-            self.pauseMutations()
-        }
-        pauseMutationsOp.addDependency(pauseSubscriptionsOp)
 
-        let setUpCloudSubscriptionsOp = CancelAwareBlockOperation {
-            self.setUpCloudSubscriptions(api: api, storageAdapter: storageAdapter)
-        }
-        setUpCloudSubscriptionsOp.addDependency(pauseMutationsOp)
-
-        let performInitialQueriesOp = CancelAwareBlockOperation {
-            self.performInitialQueries()
-        }
-        performInitialQueriesOp.addDependency(setUpCloudSubscriptionsOp)
-
-        let activateCloudSubscriptionsOp = CancelAwareBlockOperation {
-            self.activateCloudSubscriptions()
-        }
-        activateCloudSubscriptionsOp.addDependency(performInitialQueriesOp)
-
-        let startMutationQueueOp = CancelAwareBlockOperation {
-            self.startMutationQueue(api: api, mutationEventPublisher: self.mutationEventPublisher)
-        }
-        startMutationQueueOp.addDependency(activateCloudSubscriptionsOp)
-
-        let updateStateOp = CancelAwareBlockOperation {
-            Amplify.Hub.dispatch(to: .dataStore,
-                                 payload: HubPayload(eventName: HubPayload.EventName.DataStore.syncStarted))
-            self.remoteSyncTopicPublisher.send(.syncStarted)
-        }
-        updateStateOp.addDependency(startMutationQueueOp)
-
-        syncQueue.addOperations([
-            pauseSubscriptionsOp,
-            pauseMutationsOp,
-            setUpCloudSubscriptionsOp,
-            performInitialQueriesOp,
-            activateCloudSubscriptionsOp,
-            startMutationQueueOp,
-            updateStateOp
-        ], waitUntilFinished: false)
+        remoteSyncTopicPublisher.send(.storageAdapterAvailable)
+        stateMachine.notify(action: .receivedStart)
     }
 
     func submit(_ mutationEvent: MutationEvent) -> Future<MutationEvent, DataStoreError> {
         return mutationEventIngester.submit(mutationEvent: mutationEvent)
     }
 
     // MARK: - Startup sequence
-
     private func pauseSubscriptions() {
         log.debug(#function)
         reconciliationQueue?.pause()
+
+        remoteSyncTopicPublisher.send(.subscriptionsPaused)
```

Comment:
Is `reconciliationQueue.pause()` synchronous, or do we need to pass in a completion handler to `pause()`, or is that something done on a subsequent PR?

Reply:
pause is currently a synchronous operation.  If pause becomes an area where we can't reliably stop all downstream components, then this may need to become async, but for now, I believe keeping this as a synchronous operation which halts further progression of remote sync engine seems to currently work.